Remove unused signature features

Mike-M0 opened the pull request:
## Describe your changes
- only create xmldsig fragments which are listed (like other fragments)
- remove unused fields for optimization (e.g remove "shall not be used" from standard) 
removed elements are not in structure and will result in en/decoder error NOT_IMPLEMENTED when "isUsed" flag is set or element is accessed in decoder.

## Issue ticket number and link
Feature request #75

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

Commit:
Add feature from request #75:
- only create xmldsig framgnets which are listed (like other fragments)
- remove unused fields for optimization (per default remove "shall not be used" from standard)
removed elements are not in structure and will result in en/decoder error NOT_IMPLEMENTED when "isUsed" flag is set.

Signed-off-by: Michael Mezger <michael.mezger@chargebyte.com>

Mike-M0 commented:
The change is not optimal.
I needed a manual change of eventCode in xmldsigFragment en/decoder from 1 to 32 and bitcount from 1 to 6 (which are the values from before this feature), because all other eventcodes are getting optimized out.
I could not find a easy way to implement this in generator.
But with this manual fix, PnC works fine again.

barsnick commented:
Notes for us when reviewing this:

- Can this or should this be configurable (on/off)?
- Does it affect the API?
- Does it affect the ABI?
- Is it free of regressions?

barsnick commented:
For the record, from my experiments:

- All the elements disappear from the API.
- Binary code size reduction (x86_64, standard generator options), iso-2 only: 1.7 to 2.1 % (reduced from 390272 to 383568 bytes stripped, 747752 to 731944 bytes unstripped)
- Memory size reduction: from
  ISO-2 DC Struct sizes: exiDoc: 24184, V2G_Message: 24184, Header: 10360, Body: 13824
  to
  ISO-2 DC Struct sizes: exiDoc: 17064, V2G_Message: 17064, Header: 3240, Body: 13824

Our regression tests seem to pass.

barsnick commented:
If we take this in, we should also do the same optimizations for DIN and ISO 15118-20.

Applied to DIN and ISO -20:
HEAD @ d0b5310fd9d3babf0490e5dd88a284b4e4a9557a: 
```
SAP Struct sizes:
	exiDoc: 588

DIN Struct sizes:
	exiDoc: 10976
	V2G_Message: 10976
	Header: 7976
	Body: 3000

ISO-2 DC Struct sizes:
	exiDoc: 24184
	V2G_Message: 24184
	Header: 10360
	Body: 13824

ISO-20 CommonMessages Struct sizes:
	exiDoc: 318392

ISO-20 AC Struct sizes:
	exiDoc: 10920

ISO-20 DC Struct sizes:
	exiDoc: 10840
```

With a modified version of this PR, rebased on HEAD:
```
SAP Struct sizes:
	exiDoc: 588

DIN Struct sizes:
	exiDoc: 4320
	V2G_Message: 4320
	Header: 1316
	Body: 3000

ISO-2 DC Struct sizes:
	exiDoc: 17064
	V2G_Message: 17064
	Header: 3240
	Body: 13824

ISO-20 CommonMessages Struct sizes:
	exiDoc: 311280

ISO-20 AC Struct sizes:
	exiDoc: 6000

ISO-20 DC Struct sizes:
	exiDoc: 6000
```

Mike-M0 committed:
Merge branch 'EVerest:main' into main

Commit:
Fix generated eventcodes of xmlsig en/decoder functions

Remove optimization flags from default config and only add them as
comments. Remove "illegal" characters from comments.

Signed-off-by: Michael Mezger <michael.mezger@chargebyte.com>
